engine/src/main/java/org/camunda/bpm/engine/impl/cmd/AbstractSetTaskPropertyCmd.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.camunda.bpm.engine.impl.cmd;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.Serializable;
+import org.camunda.bpm.engine.impl.cfg.CommandChecker;
+import org.camunda.bpm.engine.impl.interceptor.Command;
+import org.camunda.bpm.engine.impl.interceptor.CommandContext;
+import org.camunda.bpm.engine.impl.persistence.entity.TaskEntity;
+import org.camunda.bpm.engine.impl.persistence.entity.TaskManager;
+
+/**
+ * Abstract command class, meant to encapsulate boilerplate logic for concrete commands that wish to set a property
+ * on a task and inherit authorization, multi-tenancy
+ *
+ * @param <T> the type of the value to set by this command
+ */
+public abstract class AbstractSetTaskPropertyCmd<T> implements Command<Void>, Serializable {
+
+  private final String taskId;
+  private final T value;
+
+  public AbstractSetTaskPropertyCmd(String taskId, T value) {
+    this.taskId = requireNonNull(taskId);
+    this.value = requireNonNull(value);
```

Comment:
I've updated the javadoc on the **AbstractSetTaskPropertyCmd** & respective _concretions_ to be more informative & also include the **NullValueException** that might be thrown in case of null inputs.